lib/ecmascript.ts

```diff
@@ -2920,75 +2930,71 @@ interface ToStringOptions {
   roundingMode: ReturnType<typeof ToTemporalRoundingMode>;
 }
 
+function formatAsDecimalNumber(num: number) {
+  if (num <= NumberMaxSafeInteger) return num.toString(10);
+  return JSBI.BigInt(num).toString();
```

Comment:
Do either the bigint ctor or JSBI.BigInt work properly for numbers larger than MAX_SAFE_INTEGER?

Reply:
Another maybe silly question: negative numbers? Or is this guaranteed to only receive positive numbers?

Reply:
> Do either the bigint ctor or JSBI.BigInt work properly for numbers larger than MAX_SAFE_INTEGER?

Yep, this is kinda the whole point of big integer types!

> Another maybe silly question: negative numbers? Or is this guaranteed to only receive positive numbers?

It's only called with positive values. We could perhaps add an assertion here, although not sure if that's needed. What do you think?

Reply:
An assertion seems reasonable to me.

I wasn't sure if the bigint ctor parsing a number that is larger than MAX_SAFE_INTEGER always yields the same value as parsing the equivalent base-10 string, but maybe the safe-ness only applies if you are doing operations across the boundary, not just declaring the numbers?

Reply:
`BigInt(12345678901234568000) === BigInt('12345678901234568000') // false`

Declaring Number values that are above the MAX_SAFE_INTEGER value can cause problems, so I think we should probably throw an error here and force callers to declare their values as bigints properly?

Does the upstream polyfill have this problem?

Reply:
I think the semantics here are intended to be "format the number exactly as `String(BigInt(number))` would do" because that's the only thing that doesn't produce scientific notation for large numbers. But note that this will be obsolete once we reform Duration calculations, because larger than MAX_SAFE_INTEGER will be disallowed.

Reply:
It sounds like this problem:
a) also exists upstream, so should probably be fixed there first
b) is going away with the upcoming Duration-as-mathematical-values change upstream

So I think I'll just leave it as-is for now, and then when the Duration change is ported we can decide if we need to do anything additional here in this repo.